airbyte-cdk/bulk/core/extract/src/main/kotlin/io/airbyte/cdk/read/Partitions.kt

```diff
@@ -154,3 +154,6 @@ data class PartitionReadCheckpoint(
     val opaqueStateValue: OpaqueStateValue,
     val numRecords: Long,
 )
+
+/** A [PartitionReader] with no time limit for its execution. */
+interface UnlimitedTimePartitionReader : PartitionReader
```

Comment:
instead of adding an interface, is it better to just use something like `timeout_value` in a member variable and check the value to determine if the timeout is unlimited?

Reply:
The contract for readers is that they're should be able to stop upon request (soft stop) and checkpoint at any moment.
The schedule of checkpointing is something that is configured, so allowing this to be coded will create two settings.
<img width="456" alt="image" src="https://github.com/user-attachments/assets/f7479b08-9a2f-4452-b104-62110bfb66fe" />

The exception here is the CDC reader which doesn't checkpoint mid-read.